src/gui/dialogs/addon/manager.cpp

```diff
@@ -391,6 +391,35 @@ void addon_manager::pre_show(window& window)
 	connect_signal_notify_modified(type_filter,
 		std::bind(&addon_manager::apply_filters, this));
 
+	// Language filter
+	// Prepare shown languages, source all available languages from the addons themselves
+	std::set<std::string> languages_available;
+	for(const auto& a : addons_) {
+		for (const auto& b : a.second.locales) {
+			languages_available.insert(b);
+		}
+	}
+
+	for (const auto& i: languages_available) {
+		std::string lang_code_string = langcode_to_string(i);
+		// Only show languages, which have a translation as per langcode_to_string() method
+		// Do not show tranlations with their langcode e.g. "sv_SV"
+		if (lang_code_string != "")
+			language_filter_types_.emplace_back(std::pair<int, std::string>	(language_filter_types_.size(), lang_code_string));
```

Comment:
Everything else looks good, though I would suggest one small style change here:
```cpp
if(std::string lang_code_string = langcode_to_string(i); !lang_code_string.empty()) {
	language_filter_types_.emplace_back(language_filter_types_.size(), std::move(lang_code_string));
}
```

Reply:
I'll go ahead and merge once this is done then.